server/src/com/thoughtworks/go/server/service/ArtifactsService.java

```diff
@@ -126,6 +133,19 @@ public boolean saveOrAppendFile(File dest, InputStream stream) {
         }
     }
 
```

Comment:
I see very similar code in [ArtifactsDiskCleaner#deleteOldArtifacts](https://github.com/gocd/gocd/blob/5422c89361c109eb29e6d38088e8c8f0125e285c/server/src/com/thoughtworks/go/server/service/ArtifactsDiskCleaner.java#L62). Can you try to reuse it.